src/spikeinterface/core/basesorting.py

```diff
@@ -529,6 +537,9 @@ def to_spike_vector(self, concatenated=True, extremum_channel_inds=None, use_cac
             spike_dtype = spike_dtype + [("channel_index", "int64")]
             ext_channel_inds = np.array([extremum_channel_inds[unit_id] for unit_id in self.unit_ids])
 
+        if use_cache and self._cached_spike_vector is None:
```

Comment:
I think that after a week I will find this line very confusing for me to read (now it makes sense because I have been in the discusssion and I know the context but I will forget or someone else that did not know will come along).

If I come to this function for the first time in the base class it says "custom_cache_spike_vector" I go to the implementation and is empty. What does it does? When does it do it ? I do ctrl+f in the library and It seems to sometimes (when?) have the side effect of setting a private property (`self._cached_spike_vector`) and then the function continues as it is was before. 

I think it would be more clear if the function were to return the `spike_vector` then you do whatever you need to do with it and do an early return here. That way, it would be very explicit in the code that you are overiding the construction of the spike_vector with custom logic for some cases.

Reply:
I don't like the idea of returning the spike vector, because then you need to work things out like concatenated vs not concatenated, and you duplicate the parameters and increase the complexity of the code.

Here the goal is simple: creating the cache, and then `to_spike_vector` does what it does with the cache.

Reply:
I think it's ok!